pilot/pkg/proxy/envoy/v2/rds.go

```diff
@@ -57,13 +56,10 @@ func (s *DiscoveryServer) generateRawRoutes(con *XdsConnection, push *model.Push
 	// Now validate each route
 	for _, r := range rawRoutes {
 		if err := r.Validate(); err != nil {
-			retErr := fmt.Errorf("RDS: Generated invalid route %s for node %v: %v", r.Name, con.modelNode, err)
 			adsLog.Errorf("RDS: Generated invalid routes for route:%s for node:%v: %v, %v", r.Name, con.modelNode.ID, err, r)
 			rdsBuildErrPushes.Increment()
 			// Generating invalid routes is a bug.
-			// Panic instead of trying to recover from that, since we can't
-			// assume anything about the state.
-			panic(retErr.Error())
+			// Instead of panic, which will break down the whole cluster. Just ignore it here, let envoy process it.
```

Comment:
Bad config can cause envoy to crash/go into dis-functional state. Are you sure it's better to push bad config rather than crash?

Reply:
I think so. Anything that would be caught by `.Validate()` will be rejected by Envoy calling the same `.Validate()` on its side, right?

Crashing Pilot is not acceptable -- one bad config can bring down the entire cluster

Reply:
Validate() is not checking for semantic issues. PIlot may push a corrupt listener that requires a non-existent cluster which will hose envoy.

Reply:
I am just removing the call to Validate so I don't see how that is related to this change? With or without this we are not catching semantic issues?

Reply:
Why does it say:
```
			// Panic instead of trying to recover from that, since we can't
			// assume anything about the state.
```
Is this comment not true?
Pushing invalid config to envoy just seems like a bad idea.

Reply:
That comment is removed in this PR.

See https://github.com/istio/istio/issues/17411 for some context. One random user in the cluster can make a virtual service that is invalid and break the entire cluster.

As it is we already miss 90% of invalid configs we send to envoy, because most of it has already been marshalled to Any before we get to this part.

The alternative would be to not respond to Envoy, and history has shown that is not a good idea, and can cause Envoy to get into a permanent bad state. NACKing config is part of the XDS protocol, I am not sure we need to avoid sending bad config? Do you have any references to breaking envoy by sending bad config

Reply:
Referential integrity is not part of validation in envoy. You can reference an invalid resource which will make envoy wait forever for it to arrive (e.g. route, cluster, secret, etc). Typically, you're supposed to get an oncall page when that happens ;)

Reply:
I don't understand what this has to do with this PR. We have never checked referential integrity in pilot, so that is completely unrelated to this? 

This .Validate() was doing exactly the same validation Envoy is doing (assuming protoc-gen-validate is the same for c++ and go?)

That is also no longer true, Envoy will now timeout after 15s rather than waiting forever, see initial_fetch_timeout